javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -448,7 +448,7 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
      * @return A new Map containing the given entries
      */
     public static <K, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4) {
-        return of(keyComparator, k1, v1, k2, v2, k3, v3).put(k4, v4);
```

Comment:
The previous version called TreeMap.put, which was bad for these reasons:

* TreeMap.put was called, which creates RedBlackTree-wrappers
* of(...) recursively called O(n) times of() before starting to add values

The current version

* calls put on RedBlackTree, which does not create intermediate TreeMap objects
* turns the recursive call into an iteration, which is more efficient in this case